### ISSUE TITLE
Concatenate[str, bool, Q] spuriously duplicated to Concatenate[str, bool, str, bool, Q]

### ISSUE THREAD

**Issue by andersk** [python/mypy#12734]
Mypy gives a false positive on this valid code. The error message seems to indicate that it’s spuriously duplicated the `str, bool` parameters to `str, bool, str, bool` somewhere.

```python
from typing import Callable, Concatenate, ParamSpec

P = ParamSpec("P")
Q = ParamSpec("Q")

def foo(f: Callable[P, int]) -> Callable[P, int]:
    return f

def bar(f: Callable[Concatenate[str, bool, Q], int]) -> Callable[Concatenate[str, bool, Q], int]:
    return foo(f)
```

[Mypy playground link](https://mypy-play.net/?mypy=latest&python=3.10&gist=8e14aa4bd1b441e495f99f01d95f95bb)

```
main.py:10: error: Incompatible return value type (got "Callable[[str, bool, str, bool, **Q], int]", expected "Callable[[str, bool, **Q], int]")
main.py:10: error: Argument 1 to "foo" has incompatible type "Callable[[str, bool, **Q], int]"; expected "Callable[[str, bool, str, bool, **Q], int]"
Found 2 errors in 1 file (checked 1 source file)
```

**Comment by andersk** [comment]
Trying to eta-expand `foo(f)` results in other weird “incompatible type” errors—one where the LHS and RHS display as identical, and two where the RHS displays as a *list* of types.

```python3
from typing import Callable, Concatenate, ParamSpec

P = ParamSpec("P")
Q = ParamSpec("Q")

def foo(f: Callable[P, int]) -> Callable[P, int]:
    return f

def bar(f: Callable[Concatenate[str, bool, Q], int]) -> Callable[Concatenate[str, bool, Q], int]:
    def g(s: str, b: bool, *args: Q.args, **kwargs: Q.kwargs) -> int:
        return foo(f)(s, b, *args, **kwargs)

    return g
```

```
main.py:11: error: Argument 1 to "foo" has incompatible type "Callable[[str, bool, **Q], int]"; expected "Callable[[str, bool, **Q], int]"
main.py:11: error: Argument 1 has incompatible type "str"; expected "[str, bool, **Q.args]"
main.py:11: error: Argument 2 has incompatible type "bool"; expected "[str, bool, **Q.args]"
Found 3 errors in 1 file (checked 1 source file)
```

**Comment by PIG208** [comment]
Relevant [code](https://github.com/python/mypy/blob/master/mypy/checkexpr.py#L1041-L1044) where each `ParamSpec` argument eventually gets expanded twice
